rfcs/0020-security-on-call.md

```diff
@@ -0,0 +1,165 @@
+shame to lose them.
+
+At the same time, most security patches are easy to apply:
+
+ - Most announcements to security email lists include easy-to-apply
+   patches.
+ - Well established, well funded distributions have full time
+   employees focusing on security matters. These other distributions
+   regularly publish their minimal security patches on their bug
+   trackers or in their source trees.
+ - Many security issues can be easily fixed by minor package bumps.
+
+Some of the more tricky parts are determining how to backport the
+patch, and finding build capacity for testing large rebuild changes.
+Both of these questions can easily be answered by asking more
+experienced community members for help.
+
+# Detailed design
+[design]: #detailed-design
+
+I propose we create an on-call rotation for publicly disclosed
+security issues. Each member of the team will be on call for 24 hours,
+and expected to handle every new issue which is created within their
+on-call period.
+
+The patching team will not handle issues under embargo.
+
+## Patching Team
+
+The team should be of at least 10 members, preferably over 14.
+
+Members of this team should range from new contributors looking to
+participate, to more skilled and well known contributors.
+
+### Requirements
+
+Each team member should:
+
+ - Know how patches work, or at least be willing to learn
+ - Be comfortable with Git and our backporting workflow
+ - Know their personal limits and be confident asking for help
+
+## On Call
+
+An on-call rotation system should be used or made to handle scheduling
+and informing people about their shift. A shift is 24hrs long, and
+should probably start at midnight UTC, to be equally unfair to
+everyone.
+
+### Responsibilities
+
+1. Monitor a well defined list of mailing lists for new issues.
+2. Ensure each issue is triaged and addressed if needed.
+
+#2 is a bit vaguely worded, as the person is not required to
+_actually_ fix the issue. They are allowed to delegate the patching to
+other people. However, they _are_ responsible for ensuring the issue
+is _fixed_.
+
+#### Triage and Fixing
+
+1. Check to see if the issue impacts each supported version of NixOS.
+2. Write and / or backport patches as applicable, either by version
+   bumps, large patches, or minimal patches.
+3. Prepare an advisory to send to the nix-security-announce mailing
+   list, which a member of the NixOS Security Team will send.
+
+## The Well Defined List of Mailing Lists
+
+The list should not live in the RFC documentation, but an external set
+of documentation used to document the security patching process.
+However, an initial starting list to consider:
+
+1. oss-security
+2. full-disclosure
+3. an assortment of distro advisory announcements:
+
+ - Arch
+ - Debian
+ - Gentoo
+ - Red Hat
+ - SUSE
+
+## Ensuring Complete Mailing List Coverage
+
+This is a tricky problem, and I propose that the first implemention
+be naive, simple, quick, and ugly.
```

Comment:
On IRC, tilpner said:

> a shared mail account seems brittle, too easy to accidentally mark a mail as read, without feeling responsible for it

I agree. I would love other suggestions on how to do this. However, anything with too much work seems tricky.

Maybe we could use pagerduty's email-to-page trigger to handle automatic assignment.